core/src/avm2/globals/flash/display/bitmap_data.rs

```diff
@@ -1336,11 +1336,15 @@ pub fn compare<'gc>(
     {
         other_bitmap_data
     } else {
-        // The documentation says that -1 should be returned here, but -2 is actually returned.
+        // The documentation for AVM1 says that -1 should be returned here,
+        // but -2 is actually returned.
+        // TODO: Has this been tested for AVM2?
```

Comment:
I don't think this branch can be reached in AVM2 since AVM2 enforces type-checking.

Reply:
Should I replace it with this?:

```rust
    let other_bitmap_data = args
        .get_object(activation, 0, "otherBitmapData")?
        .as_bitmap_data()
        .unwrap(); // The AVM2 enforces type-safety.
```

Reply:
No wait, can that become `null`?

Reply:
Would it be alright if I instead replace it with:

```rust

        // The documentation for AVM1 says that -1 should be returned here,
        // but -2 is actually returned.
        // TODO: For AVM2, this branch should never get reached, since
        //   AVM2 checks types.
```

? Since I would prefer not to change the behavior of that function in this PR (it might have been/be better to leave out these `TODO`s of the PR).

Reply:
Does the `TODO` look good now?